File: v1_cpp/game/DeadLine.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   DeadLine.cpp
 * Author: alex
 * 
 * Created on September 4, 2018, 10:39 PM
 */

#include "DeadLine.h"

DeadLine::DeadLine(int type, int max_length, int max_height, cpSpace *space) {
    
    mType = type;
    cpVect fp = cpv(0, 0);
    cpVect sp = cpv(max_length, 0);
    cpBody * lineBody = cpBodyNewKinematic(); //self.line_body = pymunk.Body(0, 0, pymunk.Body.KINEMATIC)
    mLine = cpSegmentShapeNew(lineBody, fp, sp, 2);
    mLine->sensor = true;
    cpBodySetPosition(mLine->body, cpv(0, mType == ASC ? 10 : max_height - 10));
}

DeadLine::DeadLine(const DeadLine& orig) {
}

DeadLine::~DeadLine() {
}

File: v1/src/main/kotlin/gameobjects/DeadLine.kt
package gameobjects

import chipmunk_bind.*


class DeadLine(val type: Int, max_length: Float, max_height: Float, space: cpSpace) {
    companion object {
        val ASC = 0
        val DESC = 1
    }

    val line = cpSegmentShape()

    init {
        val fp = cpVect(0f, 0f)
        val sp = cpVect(max_length, 0f)
        val line_body = cpBody()
        line_body.cpBodyInit(0f, 0f)
        line_body.cpBodySetType(cpBodyType.CP_BODY_TYPE_KINEMATIC)

        line.init(line_body, fp, sp, 2f)
        line.sensor = true
        line.body?.cpBodySetPosition(if (type == ASC) cpVect(0f, 10f) else cpVect(0f, max_height - 10))
    }

    fun move() {
        val position = line.body?.cpBodyGetPosition() ?: cpVect.cpvzero
        if (type == ASC) {
            line.body?.cpBodySetPosition(cpVect(position.x, position.y + 0.5f))
        } else {
            line.body?.cpBodySetPosition(cpVect(position.x, position.y - 0.5f))
        }
    }

    fun get_object_for_space(): cpSegmentShape {
        return line
    }

    fun get_position(): Float {
        return line.body?.cpBodyGetPosition()?.y?:0f
    }
}



